### DISCUSSION TITLE
For those who cant access their profile because it needs auth.

### DISCUSSION THREAD

**Discussion by apappas1129** [isturiz/sync-graph#3] [Ideas]
Because visiting https://github.com/isturiz/sync-graph/blob/ca5dcb24ea4c715287f0db4ee3096884c47da235/src/script.js#L58
didn't work for me since it requires a strict 2FA , What I did instead was manually visited my profile page and copied the entire html and saved it to an html file sibling to the `script.js` file. And then I modified the function a bit to just read the file directly instead of  using `fetchPageWithPuppeteer`.

```javascript
import { promises as fs } from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

// Get the __dirname equivalent for ESM
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

export async function generateCommitScript({ username, minYear, maxYear }) {
  const commitData = new Map();

  // Construct the path to the sibling test.html file
  const filePath = path.join(__dirname, 'test.html');

  // Read the content of the test.html file
  const hardcodedPage = await fs.readFile(filePath, 'utf8');

    const datesForYear = extractCommitDates({ html: hardcodedPage });
    datesForYear.forEach((count, date) => commitData.set(date, count));


  writeScriptFile({ commitData: commitData });
}

```

Although I am having trouble with the commit signature because even if I set the username and email identical to my personal github that I want my commits to be registered in,

```bash
git config user.name "apappas1129"
git config user.email "myemail@gmail.com"
```

it didn't actually display the commits as if with the link and photo of my profile. Instead it just looked like this:
![image](https://github.com/user-attachments/assets/02fbaf43-855b-4be7-880a-ea470f8e484c)
instead of something like this:
![image](https://github.com/user-attachments/assets/eba1a5e7-cdb5-47c5-ad62-0651f186bb27)

**Comment by apappas1129** [comment]
I opened a PR https://github.com/isturiz/sync-graph/pull/5 to address the issue described above regarding author and committer not linked to the primary account even with the correct git config.

**Reply by isturiz** [reply]
I will review the PR, and if everything looks great, I'll proceed with the merge. :)

Thank you for wanting to improve the project; I truly appreciate it.

**Reply by apappas1129** [reply]
Thank you! I initially considered adding an option for users to manually create the HTML file as I did above. However, I realized that this would introduce additional complexity to the tool, which I want to avoid. . If I find the time, I’ll explore creating a separate tool based on this repo to specifically support those manual steps only.

**Comment by apappas1129** [comment]
I just made a silly mistake on expecting the commits to be counted as contribution when pushed on the forked repo
![image](https://github.com/user-attachments/assets/f4207b09-7d71-4a5d-8d0e-91f7a6c7bda3)
It seems that to create an actual repo for this to work. Might want to indicate this in the Readme @isturiz

**Reply by isturiz** [reply]
Yes, that's exactly why I created the repository as a template, so that anyone can use it without encountering that issue.

GitHub doesn't offer many alternatives for syncing commits, and some of the solutions tend to be quite tricky. When I created the project, I tried to keep it as simple as possible and focused on what really matters; however, I think that despite being a simple solution to the initial problem, it still lacks clarity and elegance. I’m aware it can be somewhat confusing.

I have limited time available (only weekends) to dedicate to open source/personal projects; if you'd like, any improvement to the README is welcome.

I have several ideas that have yet to be developed:

- Improve the README (add clarity)
- Add unit tests
- Improve the CLI

**Reply by isturiz** [reply]
Btw, I noticed that you created a similar repository, [Git Synced](https://github.com/apappas1129/git-synced). I've added it to the README in a new section called "Similar Tools." 🙌🏻

@apappas1129
